server.js

```diff
@@ -272,17 +287,20 @@ var renderChart = (request, window, callback) => {
     .style('stroke', (d, i) => `#${lineColors[i] || 'D0D3E0'}`)
     .attr('d', (d) => line(_.zip(columns, d)));
 
-  chart.selectAll('g.circles')
+if (applyCircles==true) {
+    chart.selectAll('g.circles')
     .data(data)
     .enter().append('g')
-      .attr('class', (d, i) => `circles circles-${i}`)
-      .selectAll('circle')
-        .data(d => d)
-        .enter().append('circle')
-          .attr('r', 5)
-          .style('fill', (d, i, j) => `#${lineColors[j] || '2ba8de'}`)
-          .attr('cx', (d, i) => x(columns[i]))
-          .attr('cy', d => y(d))
+     .attr('class', (d, i) => `circles circles-${i}`)
+     .selectAll('circle')
+     .data(d => d)
+     .enter().append('circle')
+     .attr('r', 5)
+     .style('fill', (d, i, j) => `#${lineColors[j] || '2ba8de'}`)
+     .attr('cx', (d, i) => x(columns[i]))
+     .attr('cy', d => y(d))
+    }
```

Comment:
same for indentation here? the original [d3 indentation actually has a meaning via convention](https://bost.ocks.org/mike/d3/workshop/#35).